Formula/llvm.rb

```diff
@@ -108,33 +128,54 @@ class Llvm < Formula
   keg_only :provided_by_osx
 
   option :universal
-  option "with-clang", "Build the Clang compiler and support libraries"
-  option "with-clang-extra-tools", "Build extra tools for Clang"
-  option "with-compiler-rt", "Build Clang runtime support libraries for code sanitizers, builtins, and profiling"
-  option "with-libcxx", "Build the libc++ standard library"
-  option "with-lld", "Build LLD linker"
+  option "without-clang", "Do not build the Clang compiler and support libraries"
+  option "without-clang-extra-tools", "Do not build extra tools for Clang"
+  option "without-compiler-rt", "Do not build Clang runtime support libraries for code sanitizers, builtins, and profiling"
+  option "without-libcxx", "Do not build the libc++ standard library"
+  option "with-libcxxabi", "Build the libc++abi standard library"
+  option "with-libunwind", "Build the libunwind library"
+  option "without-lld", "Do not build LLD linker"
   option "with-lldb", "Build LLDB debugger"
+  option "with-openmp", "Build additional OpenMP Runtime Libraries"
   option "with-python", "Build Python bindings against Homebrew Python"
-  option "with-rtti", "Build with C++ RTTI"
-  option "with-utils", "Install utility binaries"
-
-  deprecated_option "rtti" => "with-rtti"
-
-  if MacOS.version <= :snow_leopard
+  option "without-rtti", "Build without C++ RTTI"
+  option "without-utils", "Do not install utility binaries"
+  option "without-polly", "Build without Polly optimizer"
+  option "with-test", "Build LLVM unit tests"
+  option "with-shared-libs", "Build all libs as shared instead of static (for developers)"
+  option "without-libffi", "Use libffi to call external functions from the interpreter"
+
+  depends_on "libffi" => :recommended # llvm.org/docs/GettingStarted.grml#requirements
+  depends_on "graphviz" => :optional # for the 'dot' tool (lldb)
+  depends_on "ocaml" => :optional
+
+  if build.with? "python"
+    depends_on "python"
+  elsif MacOS.version <= :snow_leopard
     depends_on :python
   else
-    depends_on :python => :optional
+    depends_on :python => :optional # not sure this is correct
   end
   depends_on "cmake" => :build
 
   if build.with? "lldb"
-    depends_on "swig"
+    depends_on "swig" if MacOS.version >= :lion
     depends_on CodesignRequirement
   end
 
   # Apple's libstdc++ is too old to build LLVM
-  fails_with :gcc
   fails_with :llvm
+  # According to the official llvm readme, GCC 4.7+ is required
+  fails_with :gcc_4_0
+  fails_with :gcc
+  ("4.3".."4.6").each do |n|
+    fails_with :gcc => n
```

Comment:
Your code is absolutely fine. Homebrew doesn't support arbitrary compilers, so `:gcc_4_0` actually means “Apple GCC 4.0”, `:gcc` means “Apple GCC 4.2”, and `:gcc => version` means the GCC installed via the corresponding Homebrew formula. I believe there was no GCC formula prior to GCC 4.3. What we're talking about is possible improvements to what `fails_with` accepts as an argument, to avoid having to specify each major GCC release individually.

Though now that I think a bit more about this, this very much looks like LLVM simply requires a compiler that supports C++11, which would mean you can replace _all_ `fails_with` lines with a [simple `needs :cxx11`](https://github.com/Homebrew/brew/blob/b05a596d570c8e5686fbb45331da8f101179f49a/Library/Homebrew/compilers.rb#L61-L69).